app/services/UserService.scala

```diff
@@ -31,12 +31,14 @@ class UserService @Inject() (implicit val connection: DBConnection) {
     Users.searchUsers(filters)
   }
 
-  def getOrCreateUserByEmail(name: String, familyName: String, email: String): Future[Int] = {
+  def getOrCreateUserByEmail(name: String, familyName: String, email: String): Future[Option[User]] = {
     Users.getUserIdByEmail(email).flatMap {
```

Comment:
We have getUserByEmail in Users, so we can have this function rewritten as follows:

Users.getUserByEmail(email).flatMap {
      case Some(user: User) =>
        Future.successful(Some(user))
      case _ =>
        val user = User(None, name, familyName, email, AccessLevel.Basic)
        Users.add(user)
        Future.successful(Some(user))
    }